backend/src/core/middlewares/campaign.middleware.ts

```diff
@@ -151,9 +153,39 @@ const isCampaignRedacted = async (
   }
 }
 
+/**
+ * Validate url origin is from Vault and check url expiry
+ */
+const isValidVaultUrl = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Response | void => {
+  try {
+    const { url } = req.body
+    const vaultUrl = new URL(url)
+    const expiry = vaultUrl.searchParams.get('Expires')
```

Comment:
Not sure if we need to explicitly check for expiry here. Perhaps we can just handle the error when we download the file from the S3 bucket later. When we try to download a file from an [expired Vault link](https://storage-test.vault.gov.sg/lennard%40open.gov.sg-req_379492e7-916e-42e9-9d98-842b316e0ef6-data_c8277f7e-52f9-4ddd-abec-3c6aebb90179?AWSAccessKeyId=AKIATNLXS5MLIQORMRW4&Expires=1616048834&Signature=7Zmstjfoi7025YIt98QxeSB5rUY%3D&response-content-disposition=attachment%3B%20filename%3D%22testpostmandataset.csv%22&datasetname), AWS will return us an error. 

The main concern is that even if we check expiry here, AWS might still throw an error when we download the file due to clock drift between our backend and S3. We might have to handle that explicitly too at the download stage.

Reply:
I see noted, in that case handling it at the download stage would be more efficient